dm/master/server.go

```diff
@@ -211,6 +232,34 @@ func errorCommonWorkerResponse(msg string, worker string) *pb.CommonWorkerRespon
 
 // StartTask implements MasterServer.StartTask
 func (s *Server) StartTask(ctx context.Context, req *pb.StartTaskRequest) (*pb.StartTaskResponse, error) {
+	log.L().Info("receive request and will save it to etcd", zap.Stringer("payload", req), zap.String("request", "StartTask"))
+
+	responseErr := func(err error) *pb.StartTaskResponse {
+		return &pb.StartTaskResponse{
+			Result: false,
+			Msg:    errors.ErrorStack(err),
+		}
+	}
+
+	reqBytes, err := req.Marshal()
+	if err != nil {
+		return responseErr(err), nil
+	}
+	responseBytes, err := s.saveRequestAndWaitResponse(ctx, pb.OperateType_StartTask, reqBytes)
+	if err != nil {
+		return responseErr(err), nil
+	}
+
+	response := &pb.StartTaskResponse{}
+	err = response.Unmarshal(responseBytes)
+	if err != nil {
+		return responseErr(err), nil
+	}
```

Comment:
1. can we define `saveRequestAndWaitResponse` as `saveRequestAndWaitResponse(ctx context.Context, tp pb.OperateType, request proto.Message, response proto.Message) error` and move `Marshal` and `Unmarshal` into it?
2. in the future, can we refactor these operation requests into one struct like
```
message OperateRequest {
    Operate op = 1;
    oneof payload {
        StartTaskRequest = 2;
        OperateTaskRequest = 3;
        ...
    }
}
```
and combine these APIs into one?

Reply:
1. I have tried it before, but `proto.Message` don't have `Marshal` function, we need to do some type assert in `saveRequestAndWaitResponse`

2. it is a good idea

Reply:
for `1`, how about `proto.Message()`?

Reply:
done in 83bc4cb